MusicCrawler.Lib/Environment/SpotifyApi.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.Diagnostics;
+using System.Text;
+using System.Web;
+using Newtonsoft.Json;
+
+namespace MusicCrawler.Lib.Environment;
+
+/**
+ * docs:
+ *      https://developer.spotify.com/documentation/web-api/reference/get-recommendations
+ *      https://developer.spotify.com/documentation/web-api/concepts/authorization
+ *      https://developer.spotify.com/documentation/web-api/tutorials/code-flow
+ *      https://developer.spotify.com/documentation/web-api/tutorials/client-credentials-flow
+ */
+public class SpotifyApi
+{
+    private readonly HttpClient _httpClient;
+
+    private readonly string _baseUri;
+
+    // The client ID from https://developer.spotify.com/dashboard
+    private const string client_id = "267c94026025449b8013ddde6d959e13";
+
+    // The client secret from https://developer.spotify.com/dashboard
+    private const string client_secret = "92c88db9315545e38989ca8cc4cad2ad";
+
+    // This is the page that the user will be sent to after clicking "accept" from the html given by accounts.spotify.com/authorize.
+    private const string redirect_uri = "http://localhost/";
+
+    public SpotifyApi()
+    {
+        this._baseUri = "https://api.spotify.com";
+        this._httpClient = new HttpClient();
+    }
+
+    public SpotifyApi(string baseUri)
+    {
+        this._baseUri = baseUri;
+        this._httpClient = new HttpClient();
+    }
+
+
+    /**
+     * responds with an html for a page where the user can click "accept".
+     * Might be useful if we want to access information specific to a spotify user.
+     */
+    public async Task<string> OAuthLogin()
+    {
+        var state = RandomStringGenerator.GenerateRandomString(16);
```

Comment:
Is this random state doing anything for us?  or mainly for testing just to see if it comes through the other side?